File: src/main/java/rustleund/fightingfantasy/framework/closures/RemoveItemClosure.java
/*
 * Created on Oct 9, 2005
 */
package rustleund.fightingfantasy.framework.closures;

import org.w3c.dom.Element;

import rustleund.fightingfantasy.framework.AbstractCommand;
import rustleund.fightingfantasy.framework.GameState;

/**
 * @author rustlea
 */
public class RemoveItemClosure extends AbstractCommand {

	private int itemId = 0;

	public RemoveItemClosure(Element element) {
		this.itemId = Integer.parseInt(element.getAttribute("id"));
	}

	@Override
	public boolean execute(GameState gameState) {
		gameState.getPlayerState().removeOneOfItem(itemId);
		return true;
	}

}

File: src/main/java/rustleund/fightingfantasy/framework/closures/impl/RemoveItemClosure.kt
/*
 * Created on Oct 9, 2005
 */
package rustleund.fightingfantasy.framework.closures.impl

import org.w3c.dom.Element
import rustleund.fightingfantasy.framework.base.GameState
import rustleund.fightingfantasy.framework.base.optionalIntAttribute
import rustleund.fightingfantasy.framework.closures.Closure

/**
 * @author rustlea
 */
class RemoveItemClosure(element: Element) : Closure {

    private val itemId = element.getAttribute("id").toInt()
    private val amount = element.optionalIntAttribute("amount") ?: 1

    override fun execute(gameState: GameState): Boolean {
        repeat(amount) {
            gameState.playerState.removeOneOfItem(itemId)
        }
        return true
    }
}